integrations/community/cloudflare-agents/typescript/examples/worker/src/agent.ts

```diff
@@ -0,0 +1,30 @@
+import { Agent } from "agents";
+
+// Minimal AG-UI event set over WS (string JSON frames)
+type Event = { type: string; [k: string]: unknown };
+const send = (ws: WebSocket, e: Event) => ws.send(JSON.stringify(e));
+
+// Minimal echo agent that emits AG-UI lifecycle:
+// RUN_STARTED -> TEXT_MESSAGE_START -> ...CONTENT... -> TEXT_MESSAGE_END -> RUN_FINISHED
+export class MyAgent extends Agent {
+  onConnect(ws: WebSocket) {
+    // harmless hello for sanity (AG-UI will ignore unknown events)
+    send(ws, { type: "READY" });
+  }
+
+  onMessage(ws: WebSocket, raw: string | ArrayBuffer) {
+    const input = typeof raw === "string" ? raw : new TextDecoder().decode(raw);
+    const thread_id = crypto.randomUUID();
+    const run_id = crypto.randomUUID();
+
+    send(ws, { type: "RUN_STARTED", thread_id, run_id });
```

Comment:
If I am not mistaken, `threadId` and `runId` should be `camelCase`.

https://docs.ag-ui.com/sdk/js/core/events#lifecycle-events